### PR TITLE
Remove Hail2 HailContext, do automatic or explicit init instead.

### PR DESCRIPTION
**Pull request by tpoterba**
All import methods moved to methods.

### CODE REVIEW

**Comment by tpoterba** [comment]
Instead of creating a hail context, I've exposed the top level `init` and `stop` methods in hail2. 

Env.hc() will call init if it's not been called yet, and print a message about initializing with default params.

File: python/hail/methods/impex.py [hunk 1]
```diff
@@ -521,3 +529,135 @@ def import_fam(path, quant_pheno=False, delimiter=r'\\s+', missing='NA'):
...
+@typecheck(populations=integral,
+           samples=integral,
+           variants=integral,
+           num_partitions=nullable(integral),
+           pop_dist=nullable(listof(numeric)),
+           fst=nullable(listof(numeric)),
+           af_dist=oneof(UniformDist, BetaDist, TruncatedBetaDist),
+           seed=integral,
+           reference_genome=nullable(GenomeReference))
+def balding_nichols_model(populations, samples, variants, num_partitions=None,
```
Comment (danking) [inline python/hail/methods/impex.py]: feels like a `statgen.py` thing

File: python/hail2/__init__.py [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
+                  master=nullable(strlike),
+                  local=strlike,
+                  log=strlike,
+                  quiet=bool,
+                  append=bool,
+                  min_block_size=integral,
+                  branching_factor=integral,
+                  tmp_dir=strlike,
+                  default_reference=strlike)
+def init(sc=None, app_name="Hail", master=None, local='local[*]',
```
Comment (cseed) [inline python/hail2/__init__.py]: I'm not sure about names as common as stop and init in the top-level scope.  But we're going to switch to `import hail as hl` and `hl.init()` soon?

File: python/hail/methods/impex.py [hunk 3]
```diff
@@ -521,3 +529,135 @@ def import_fam(path, quant_pheno=False, delimiter=r'\\s+', missing='NA'):
...
+@typecheck(path=oneof(strlike, listof(strlike)),
+           force=bool,
+           force_bgz=bool,
+           header_file=nullable(strlike),
+           min_partitions=nullable(integral),
+           drop_samples=bool,
+           call_fields=oneof(strlike, listof(strlike)),
+           reference_genome=nullable(GenomeReference),
+           contig_recoding=nullable(dictof(strlike, strlike)))
+def import_vcf(path, force=False, force_bgz=False, header_file=None, min_partitions=None,
```
Comment (cseed) [inline python/hail/methods/impex.py]: I still think methods.import_vcf still reads strange, but I can get used to it.  But we're going to import of methods into toplevel scope when we switch to `import hail as hl`, right?  Then I'm OK.

File: python/hail2/__init__.py (hunk 2 above)
Comment (tpoterba) [reply python/hail2/__init__.py]: yes. this is a small step to getting there

File: python/hail/methods/impex.py (hunk 3 above)
Comment (tpoterba) [reply python/hail/methods/impex.py]: yes, this will be `hl.import_vcf`

**Pull request merged**